006-vortexa-tech-challenge/src/map.tsx

```diff
@@ -0,0 +1,87 @@
+import "maplibre-gl/dist/maplibre-gl.css";
+
+import { Feature, MultiPolygon } from "geojson";
+import { RampData, RampProperties } from "./types";
+import React, { Dispatch, SetStateAction, useState } from "react";
+import ReactMap, { Marker, Popup, ViewStateChangeEvent } from "react-map-gl";
+
+import maplibregl from "maplibre-gl";
+
+export const Map = ({
+  ramps,
+  visibleFeatures,
+  setVisibleFeatures,
+}: {
+  ramps: RampData;
+  visibleFeatures: Feature<MultiPolygon, RampProperties>[] | [];
+  setVisibleFeatures: Dispatch<
+    SetStateAction<Feature<MultiPolygon, RampProperties>[] | []>
+  >;
+}) => {
```

Comment:
Let's make this a little readable:
1. Define the interface for Map Props. For now define the type in the same file, but you can use separate file aswell
2. Access props through object accessors instead of restructuring, to leverage from lints

```suggestion
type VisibleFeature = Feature<MultiPolygon, RampProperties>

interface MapProps {
  ramps: RampData;
  visibleFeatures: VisibleFeature[] | [];
  setVisibleFeatures: Dispatch<SetStateAction<VisibleFeature[] | []>>;
}

export const Map = (props: MapProps) => {
```